Bottom: 5a1ad53df84ac6d072d20d84719ea746131f9bb8
Top:    5f6c6b966220e76edea9d1c602091698753840b1
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-03-18 20:55:38 +0000

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index 679c32d..2e8444e 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -230,6 +230,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
